### PR TITLE
Feat/pending votes indicator

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
- Adds indicator for pending votes
- When opening Voting tab, latest round will show
Screenshot 2024-01-22 at 14 44 35

---

### Detailed summary
- Added a new GraphQL query `useVotingHistory` in `web/src/hooks/queries/useVotingHistory.ts` to fetch voting history data.
- Added a new utility function `getVoteChoice` in `web/src/utils/getVoteChoice.ts` to get the vote choice based on the selected answer.
- Updated the styling of `InfoContainer` component in `web/src/components/InfoCard.tsx`.
- Updated the import statements and components in `web/src/components/Verdict/DisputeTimeline.tsx`.
- Added a new utility function `getDrawnJurorsWithCount` in `web/src/utils/getDrawnJurorsWithCount.ts` to get the drawn jurors with their vote count.
- Added a new component `AccordionTitle` in `web/src/pages/Cases/CaseDetails/Voting/VotesDetails/AccordionTitle.tsx` to display the accordion title in voting details.
- Added a new component `VotesAccordion` in `web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx` to display the votes accordion in voting details.
- Updated the `VotingHistory` component in `web/src/pages/Cases/CaseDetails/Voting/VotingHistory.tsx` to fetch and display the voting history data.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by alcercu — DISMISSED** [review]
I like it on desktop, but on mobile looks awkward 🤔 
image

What if on mobile we leave only the address  on the title and move back into the accordion the choice + the number of votes? 🤔

**Review by kemuru — CHANGES_REQUESTED** [review]
I prefer the Klerosboard approach where we don't put the word "Voted:" before the vote itself on the preview.

image

then when we extend we show "Voted:"

image

and to fix the mobile looking weird, we could distribute the content in 3 different rows: 1st row is wallet address and icon, 2nd row is vote, 3rd row is number of votes. and then extended, maybe add the date of the vote too like in Klerosboard

**Comment by kemuru** [comment]
> I like it on desktop, but on mobile looks awkward 🤔 image
> 
> What if on mobile we leave only the address on the title and move back into the accordion the choice + the number of votes? 🤔

we could distribute the content in 3 different rows: 1st row is wallet address and icon, 2nd row is vote, 3rd row is number of votes. and then extended, maybe add the date of the vote too like in Klerosboard

File: web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+const AccordionContent: React.FC<{
+  choice?: string;
+  answers: Answer[];
+  justification: string;
+}> = ({ justification, choice, answers }) => (
+  <JustificationContainer>
+    {!isUndefined(choice) ? (
+      <StyledLabel>
+        Voted : <small>{getVoteChoice(parseInt(choice), answers)}</small>
```
Comment (alcercu) [inline web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx:49]: I think I am with @kemuru on this, let's remove the `Voted :` and leave only the choice
```suggestion
        <small>{getVoteChoice(parseInt(choice), answers)}</small>
```

File: web/src/pages/Cases/CaseDetails/Voting/VotesDetails/AccordionTitle.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+`;
+
+const VoteStatus: React.FC<{
+  choice?: string;
+  period: string;
+  answers: Answer[];
+  isActiveRound: boolean;
+}> = ({ choice, period, answers, isActiveRound }) => {
+  if (isUndefined(choice) && (isActiveRound ? ["appeal", "execution"].includes(period) : true))
+    return <StyledLabel>Forgot to vote</StyledLabel>;
```
Comment (alcercu) [inline web/src/pages/Cases/CaseDetails/Voting/VotesDetails/AccordionTitle.tsx:40]: Maybe change this to:
```suggestion
    return <StyledLabel>Didn't cast a vote</StyledLabel>;
```
I think this is more explicative.

**Review by alcercu — COMMENTED** [review]
Let's go with the 3 row approach on mobile, but let's remove the `Voted:` prefix to the choice

File: web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx (hunk 1 above)
Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx:49]: The "Voted :" here is the one in extended accordion, on the title "Voted :" has already been removed.

Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/Voting/VotesDetails/index.tsx:49]: Screenshot 2024-01-23 at 23 03 19
 final look

**Review by jaybuidl — APPROVED** [review]

**Pull request merged**